src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1209,7 +1209,7 @@ bool Arguments::process_settings_file(const char* file_name, bool should_exist,
       } else if (in_quote && (c == quote_c)) {
         in_quote = false;
       } else {
-        token[pos++] = c;
+        token[pos++] = checked_cast<char>(c);
       }
     }
     c = getc(stream);
```

Comment:
How about int c0 for the getc value, then the first line in the loop does
char c = checked_cast<char>(c0);
No other changes should be needed.

Reply:
There was a reason that c had to be an int.  Now I'll have to rebuild and figure it out again.

Reply:
No, it works and is a bit better, so I made this change.

Reply:
I see error in some builds due to the EOF comparison though now that c is a char. I think Dean was suggesting doing the case inside the loop.

Reply:
That must have been the error I was seeing but couldn't remember.  I'm putting this back to the version I tested completely.  I don't see the point of adding another local variable here and doing the checked_cast<> until we need the char to be stored.

Reply:
`getc` returns an int because it has to be able to return all valid char values plus EOF.